appserver/web/web-glue/src/main/java/com/sun/enterprise/web/WebDeployer.java

```diff
@@ -136,6 +136,16 @@ public <V> V loadMetaData(Class<V> type, DeploymentContext dc) {
             if (contextRoot == null)
                 contextRoot = ((GenericHandler)dc.getArchiveHandler()).getDefaultApplicationNameFromArchiveName(dc.getOriginalSource());
 
+
+            if (this.env.getRuntimeType().isMicro()) {
```

Comment:
FYI: I see this kind of code more and more sprinkled throughout the codebase.
This is a code smell. There should be a strategy pattern utilized here to get more cohesion / convergence of code that belongs in Micro / Server / DAS / instance